app/models/room.rb

```diff
@@ -31,10 +31,16 @@ def anyone_joins_as_moderator?
   end
 
   def viewer_access_code
+    config = MeetingOption.get_config_value(name: 'glViewerAccessCode', provider: 'greenlight')&.value
+    return nil if config == 'false'
+
     get_setting(name: 'glViewerAccessCode')&.value
   end
 
   def moderator_access_code
+    config = MeetingOption.get_config_value(name: 'glModeratorAccessCode', provider: 'greenlight')&.value
+    return nil if config == 'false'
+
     get_setting(name: 'glModeratorAccessCode')&.value
```

Comment:
Wouldn't you just call the RoomSettingGetter instead?

Reply:
Because this is a config **not** a room list of settings.
- `RoomSettingGetter` will return a `Hash(<name> => <value>)` of a given room settings values while taking into consideration the available rooms configurations.
- `MeetingOption.get_config_value` which was introduced on #3713 will return the value of a rooms configuration i.e. a forced enabled ``"record"` configuration.

Reply:
At the moment of this writing, we don't have a `RoomSettingGetter` that fetches by name a room setting.
We have a `RoomSettingsGetter` service that fetches a room `Hash(<name> => <value>)` of settings while considering the application rooms configuration and another service `SettingGetter` that returns the value for a given site setting while binarizing it.

Having a service that fetches only a room setting does not have to be on a service since it does not exceed in terms of functionality what is handled by the data layer.

Isn't the same reason why we removed the `RoomConfigGetter` service in favor of `MeetingOption.get_config_value`?